Tests/SonarScanner.MSBuild.PreProcessor.Test/Infrastructure/MockRoslynAnalyzerProvider.cs

```diff
@@ -27,17 +27,15 @@
 
 namespace SonarScanner.MSBuild.PreProcessor.Test;
 
-internal class MockRoslynAnalyzerProvider : RoslynAnalyzerProvider
+internal class MockRoslynAnalyzerProvider(BuildSettings buildSettings, IAnalysisPropertyProvider sonarProperties, IEnumerable<SonarRule> rules, string language) : RoslynAnalyzerProvider(new MockAnalyzerInstaller(), new TestLogger(), buildSettings, sonarProperties, rules, language)
 {
     public AnalyzerSettings SettingsToReturn { get; set; }
 
     public IAnalysisPropertyProvider SuppliedSonarProperties { get; private set; }
 
-    public MockRoslynAnalyzerProvider() : base(new MockAnalyzerInstaller(), new TestLogger()) {}
-
-    public override AnalyzerSettings SetupAnalyzer(BuildSettings buildSettings, IAnalysisPropertyProvider sonarProperties, IEnumerable<SonarRule> rules, string language)
+    public override AnalyzerSettings SetupAnalyzer()
     {
-        buildSettings.Should().NotBeNull();
+        teamBuildSettings.Should().NotBeNull();
```

Comment:
None of these assertion makes sense, as there are null checks in the base. Just remove them